src/codemodder/codeql.py

```diff
@@ -80,3 +80,12 @@ def from_sarif(cls, sarif_file: str | Path, truncate_rule_id: bool = False) -> S
                     )
                     result_set.add_result(codeql_result)
         return result_set
+
+
+# TODO: cache, make hashable
```

Comment:
I don't have a ton of bandwidth so left it as a todo, since we're passing dict of lists and some other unhashable data here, it's not super fast to do, but we should do it at some point